src/cli/commands/retrieve_history.go

```diff
@@ -129,6 +132,9 @@ func RetrieveHistory(cliConnection api.Connection, appName string, startTime, en
 		}
 
 		if !next || currentNumber >= recordNumber {
+			if next && recordNumber == 0 {
+				moreResult = true
+			}
```

Comment:
the two "if" statement here  are too tricky and complex.  Could you please simply it to make it easier for understanding?